components/tidb_query_datatype/src/codec/data_type/mod.rs

```diff
@@ -257,7 +257,39 @@ macro_rules! impl_evaluable_type {
     };
 }
 
-impl_evaluable_type! { Int }
+impl Evaluable for Int {
+    const EVAL_TYPE: EvalType = EvalType::Int;
+
+    #[inline]
+    fn borrow_scalar_value(v: &ScalarValue) -> Option<&Self> {
+        match v {
+            ScalarValue::Int(x) => x.as_ref(),
+            ScalarValue::Enum(x) => x
+                .as_ref()
+                .map(|x| unsafe { std::mem::transmute(x.value_ref()) }),
```

Comment:
Why do we need to use `std::men::transmute` here? This should be able to compile without this transmute.

Reply:
I thought `&u64` cannot be converted into `&i64` in safe mode.